users/src/test/java/io/crate/integrationtests/BaseUsersIntegrationTest.java

```diff
@@ -45,12 +49,37 @@
     protected SQLOperations.Session createUserSession() {
         return createUserSession(null);
     }
+    protected SQLOperations.Session createDDLUserSession() {
+        return createDDLUserSession(null);
+    }
+    protected SQLOperations.Session createDMLUserSession() {
+        return createDMLUserSession(null);
+    }
+    protected SQLOperations.Session createDQLUserSession() {
+        return createDQLUserSession(null);
+    }
 
     SQLOperations.Session createUserSession(String node) {
         SQLOperations sqlOperations = internalCluster().getInstance(SQLOperations.class, node);
         return sqlOperations.createSession(null, new User("normal", ImmutableSet.of(), ImmutableSet.of()), Option.NONE, DEFAULT_SOFT_LIMIT);
     }
 
+
+    SQLOperations.Session createDDLUserSession(String node) {
+        SQLOperations sqlOperations = internalCluster().getInstance(SQLOperations.class, node);
+        return sqlOperations.createSession(null, new User("ddlUser", ImmutableSet.of(), ImmutableSet.of(new Privilege(Privilege.State.GRANT, Privilege.Type.DDL, Privilege.Clazz.CLUSTER, null, "crate"))), Option.NONE, DEFAULT_SOFT_LIMIT);
+    }
+
+    SQLOperations.Session createDMLUserSession(String node) {
+        SQLOperations sqlOperations = internalCluster().getInstance(SQLOperations.class, node);
+        return sqlOperations.createSession(null, new User("dmlUser", ImmutableSet.of(), ImmutableSet.of(new Privilege(Privilege.State.GRANT, Privilege.Type.DML, Privilege.Clazz.CLUSTER, null, "crate"))), Option.NONE, DEFAULT_SOFT_LIMIT);
+    }
+
+    SQLOperations.Session createDQLUserSession(String node) {
```

Comment:
can we replace all the concrete impl here with just a generic one, taking a userName and a set of privileges?